nixos/modules/services/networking/atticd.nix

```diff
@@ -0,0 +1,241 @@
+        default = null;
+      };
+
+      user = lib.mkOption {
+        description = ''
+          The group under which attic runs.
+        '';
+        type = types.str;
+        default = "atticd";
+      };
+
+      group = lib.mkOption {
+        description = ''
+          The user under which attic runs.
+        '';
+        type = types.str;
+        default = "atticd";
+      };
+
+      settings = lib.mkOption {
+        description = ''
+          Structured configurations of atticd.
+          See https://github.com/zhaofengli/attic/blob/main/server/src/config-template.toml
+        '';
+        type = format.type;
+        default = { };
+      };
+
+      mode = lib.mkOption {
+        description = ''
+          Mode in which to run the server.
+
+          'monolithic' runs all components, and is suitable for single-node deployments.
+
+          'api-server' runs only the API server, and is suitable for clustering.
+
+          'garbage-collector' only runs the garbage collector periodically.
+
+          A simple NixOS-based Attic deployment will typically have one 'monolithic' and any number of 'api-server' nodes.
+
+          There are several other supported modes that perform one-off operations, but these are the only ones that make sense to run via the NixOS module.
+        '';
+        type = lib.types.enum [
+          "monolithic"
+          "api-server"
+          "garbage-collector"
+        ];
+        default = "monolithic";
+      };
+    };
+  };
+
+  config = lib.mkIf (cfg.enable) (
+    lib.mkMerge [
+      {
+        assertions = [
+          {
+            assertion = cfg.environmentFile != null;
+            message = ''
+              <option>services.atticd.environmentFile</option> is not set.
+
+              Run `openssl genrsa -traditional 4096 | base64 -w0` and create a file with the following contents:
+
+              ATTIC_SERVER_TOKEN_RS256_SECRET="output from command"
+
+              Then, set `services.atticd.environmentFile` to the quoted absolute path of the file.
+            '';
+          }
+        ];
+
+        services.atticd.settings = {
+          chunking = lib.mkDefault {
+            nar-size-threshold = 65536;
+            min-size = 16384; # 16 KiB
+            avg-size = 65536; # 64 KiB
+            max-size = 262144; # 256 KiB
+          };
+
+          database.url = lib.mkDefault "sqlite:///var/lib/atticd/server.db?mode=rwc";
+
+          # "storage" is internally tagged
+          # if the user sets something the whole thing must be replaced
+          storage = lib.mkDefault {
+            type = "local";
+            path = "/var/lib/atticd/storage";
+          };
+        };
+
+        systemd.services.atticd = {
+          wantedBy = [ "multi-user.target" ];
+          after =
+            [ "network.target" ]
+            ++ lib.optionals hasLocalPostgresDB [
+              "postgresql.service"
+              "nss-lookup.target"
```

Comment:
i know that upstream sets the same, but i dont really see why we need nss just in case we use a local postgres but no remote database.
Isnt nss-lookup.target making sure we can resolve hostnames? if i remember that correctly i think it make sense to run always after nss-lookup.target and not just when we are using a local postgres.

Reply:
I dropped the nss-lookup-target and made after depend on network-online.target. Furthermore, I also added a requires for postgresql if a local db.